dev_tools/Dockerfile

```diff
@@ -1,8 +1,18 @@
-FROM ubuntu
+FROM python:3.7-slim AS compile-image
 
 # Install dependencies.
+# rm -rf /var/lib/apt/lists/* cleans up apt cache. See https://docs.docker.com/develop/develop-images/dockerfile_best-practices/
 RUN DEBIAN_FRONTEND=noninteractive apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y \
-     python3-pip python3-tk texlive-latex-base latexmk git emacs vim locales
+     python3-pip \
```

Comment:
do we still need pip? python 3.7-slim doesn't come with it?

Reply:
Good call.

Reply:
Turns out that we need pip bc it includes gcc, which is required to install the immutables package (dependency of pyquil) and cytoolz

Reply:
I think this is fine for now.